auth/src/main/java/org/aerogear/mobile/security/AsyncSecurityCheckExecutor.java

```diff
@@ -0,0 +1,41 @@
+package org.aerogear.mobile.security;
+
+import android.content.Context;
+
+import org.aerogear.mobile.core.metrics.MetricsService;
+import org.aerogear.mobile.security.metrics.SecurityCheckResultMetric;
+
+import java.util.Collection;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+
+public class AsyncSecurityCheckExecutor extends AbstractSecurityCheckExecutor {
+
+    private final ExecutorService pool = Executors.newFixedThreadPool(10);
```

Comment:
the core module provides AppExecutor, it's best to keep using it.

Reply:
Also not sure if we actually need a pool of executors. For the security checks, I think it's ok for them to all happen on one background thread. Using a pool seems a little bit too much.

Reply:
Yeah, was going to add a method to set the executor in the builder